### DISCUSSION TITLE
Generate json/yaml data

### DISCUSSION THREAD

**Discussion by TheNetAdmin** [lowlighter/metrics#543] [Ideas]
I think it would be great if metrics can generate a json or yaml data file containing all the info used to generate the svg file. In this way, we can reuse metrics' data for other purposes.

I guess there is such a data structure when metrics is generating the svg file, so maybe just dump that into a json file? This is just my elementary guess.

**Comment by lowlighter** [comment]
It's already possible to output json data 🙂 
You can even test it online like [metrics.lecoq.io/github-username?config.output=json](https://metrics.lecoq.io/octocat?config.output=json)

Here's all supported output formats:
https://github.com/lowlighter/metrics/blob/f38f905ba9b727861cdeb00617dce065ab24f3eb/source/plugins/core/metadata.yml#L194-L206

Note that resulting json scheme is not always consistant between versions (some stuff get moved around) so be sure to tag a version when reusing these data somewhere else.

Also, the output still depends on which plugins are enabled, so if you want to collect data from a specific plugin, it is still required to enable it and to configure it

**Reply by TheNetAdmin** [reply]
Aha, this is awesome! Didn't know that. Great work!
